pyscada/visa/devices/Keithley_DMM2000.py

```diff
@@ -1,36 +1,101 @@
-
 from pyscada.visa.devices import GenericDevice
+from pyscada.models import DeviceWriteTask, Variable, Device
+from pyscada.models import RecordedData
+import time
+import math
+
+import logging
 
+logger = logging.getLogger(__name__)
 
 class Handler(GenericDevice):
-    """
+    '''
     Keithley DMM 2000 and other Devices with the same command set
-    """
-    
+    '''
+
     def read_data(self,device_property):
-        """
+        '''
         read values from the device
-        """
+        '''
         if self.inst is None:
-             return
+            logger.error("Visa-Keithley-read data-Self.inst : None")
+            return None
         if device_property == 'present_value':
-            return self.parse_value(self.inst.query(':FETCH?'))
+            logger.info("Keithley READ Handler")
+            return self.parse_value(self.inst.query(':READ?'))
+        else:
+            value = self.inst.query(device_property)
+            logger.info("Visa-Keithley-read data-property : %s - value : %s" %(device_property, value))
+            return self.parse_value(value)
         return None
-    
+
     def write_data(self,variable_id, value):
-        """
+        '''
         write values to the device
-        """
-        return False
+        '''
+        i=0
+        j=0
+        while i<10:
+            try:
+                self.inst.query('*IDN?')
+                #logger.info("Visa-Keithley-Write- variable_id : %s et value : %s" %(variable_id, value))
+                i=12
+                j=1
+            except:
+                time.sleep(1)
+                i += 1
+                logger.error("Keithley connect error i : %s" %i)
+        if j == 0:
+            logger.error("Keithley-Instrument non connecté")
+            return False
+        if variable_id == 'present_value':
+            i = 0
+            while i<10:
+                T1=time.time()
+#                self.inst.timeout(5)
+#                res=self.inst.timeout
+#                logger.info("timeout : %s" %res)
+#                self.inst.write(':READ?')
+#                time.sleep(0.1)
+#                Vseff = self.parse_value(self.inst.read(termination='\n'))
+#                Vseff = self.parse_value(self.inst.read())
+                self.inst.read_termination = '\n'
+                Vseff = ""
+                try:
+                    Vseff = self.parse_value(self.inst.query(':READ?'))
+                    T2=time.time()
+                except:
+                    Vsedd = ""
+                if Vseff is None or Vseff is "":
+                    i += 1
+                    logger.error("Keithley - Error Read - i : %s" %i)
+                    self.inst.write('*CLS')
+                else:
+                    i = 12
+                    #Call Phase Osc
+                    T3=time.time()
+                    cwt = DeviceWriteTask(variable_id=Variable.objects.get(name='Find_Phase_Osc').id, value=Vseff, start=time.time())
```

Comment:
i have no better solution at the moment but i'am not happy with the reference to variable.name